### PR TITLE
SC-1259

### PR DESCRIPTION
**Pull request by anmol2302**
<b>Ticket Ref:</b> [SC-1259](https://project-sunbird.atlassian.net/browse/SC-1259)
1: Doc Link : [Certificate Add Doc](https://project-sunbird.atlassian.net/wiki/spaces/User/pages/1066762497/Ecreds+-+Phase3+and+SB+rollout#Ecreds-Phase3andSBrollout-1.POSTprivate/user/v1/certs/add(PrivateAPI,willnotrequireauthtoken))

1: All the Mandatory Params being checked in the Validation 
2: Format of jsonUrl and pdfUrl is also being checked at time of Validation using class UrlValidator()(org.apache.commons.validator.UrlValidator)
3: if the URL found is not valid will throw an exception of invalid Url.

### CODE REVIEW

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/certificatevalidator/CertAddRequestValidator.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    private void validateUrl(String url){
+        if(!urlValidator.isValid(url)){
+            ProjectCommonException.throwClientErrorException(ResponseCode.invalidUrl);
+        }
+    }
+    private void validateUrls(){
+        validateUrl((String) request.get(JsonKey.PDF_URL));
+        validateUrl((String)request.get(JsonKey.JSON_URL));
```
Comment (manzarul) [inline sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/certificatevalidator/CertAddRequestValidator.java]: json-url is not url

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java [hunk 2]
```diff
@@ -442,6 +442,7 @@
     String TO_ACCOUNT_ID_MISSING = "To Account id is mandatory.";
     String FROM_ACCOUNT_ID_NOT_EXISTS = "From Account id not exists";
     String PARAM_NOT_MATCH="%s-NOT-MATCH";
+    String INVALID_PROVIDED_URL="PLEASE PROVIDE VALID URL";
```
Comment (manzarul) [inline sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java]: Message should not be same as key, see above message

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java [hunk 3]
```diff
@@ -820,5 +821,6 @@
     String TO_ACCOUNT_ID_MISSING = "TO_ACCOUNT_ID_MISSING";
     String FROM_ACCOUNT_ID_NOT_EXISTS = "FROM_ACCOUNT_ID_NOT_EXISTS";
     String PARAM_NOT_MATCH= "%s-NOT-MATCH";
+    String INVALID_PROVIDED_URL="INVALID_PROVIDED_URL";
```
Comment (manzarul) [inline sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java]: INVALID_PROVIDED_URL to INVALID_URL

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/BaseRequestValidator.java [hunk 4]
```diff
@@ -151,6 +151,39 @@ public void checkMandatoryFieldsPresent(Map<String, Object> data, String... keys
...
+   * @param mandatoryParamsList List of string represents the mandatory fields.
+   */
+  public void checkMandatoryFieldsPresent(Map<String, Object> data, List<String>mandatoryParamsList) {
+    if (MapUtils.isEmpty(data)) {
+      throw new ProjectCommonException(
+              ResponseCode.invalidRequestData.getErrorCode(),
+              ResponseCode.invalidRequestData.getErrorMessage(),
+              ResponseCode.CLIENT_ERROR.getResponseCode());
+    }
+    mandatoryParamsList
```
[the remark this answers is not in the document]
Comment (anmol2302) [reply sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/BaseRequestValidator.java]: done

**Pull request merged**